### PR TITLE
[build] Pair collaboration

### PR DESCRIPTION
**Pull request by lina**

### CODE REVIEW

**Comment by linalu1** [comment]
Ready to be reviewed/merged

**Comment by AndrewKishino** [comment]
lgtm

File: client/app/collab/collab.js [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+  // }
+  
+  $scope.sendUserTextChangeSocketSignal = function() {
+    $scope.userEditorProperties = {
+      cursorCoords: $scope.userEditor.cursorCoords(), 
+      editorValue: $scope.userEditor.getValue(), 
+      cursorPosition: $scope.userEditor.getCursor()
+    };
+    $rootScope.collabSocket.emit('userTextChange', $scope.userEditorProperties);
+  }
```
Comment (codydaig) [inline client/app/collab/collab.js]: Semi Colon

File: server/config/collabHandler.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    });
+
+
+    socket.on('getOpponent', function(data){
+      socket.broadcast.to(joinedRoom.id).emit('nameReq');
+    });
+
+    socket.on('nameSend', function(data){
+      socket.broadcast.to(joinedRoom.id).emit('opponentReady', data);
+    })
```
Comment (codydaig) [inline server/config/collabHandler.js]: Semi Colon

File: server/config/middleware.js [hunk 3]
```diff
@@ -18,6 +20,10 @@ module.exports = function (app, express) {
   app.use('/api/users', userRouter);
   app.use('/api/duels', duelRouter);
   app.use('/api/battles', battleRouter);
+  app.use('/api/collab', collabRouter);
+  app.use('/api/collabs', collabingRouter);
+
+
```
Comment (codydaig) [inline server/config/middleware.js]: Extra white space?

**Comment by codydaig** [comment]
@linalu1 Made some small syntax notes. Fix those and it will be good.

**Comment by lina** [comment]
@Kevbyte  please give seal of approval

**Comment by Kevbyte** [comment]
LGTM

**Pull request merged**